code/game/objects/structures/crates_lockers/closets/utility_closets.dm

```diff
@@ -67,38 +65,24 @@
 	desc = "It's a storage unit for tools."
 	closet_appearance = /decl/closet_appearance/secure_closet/engineering/tools
 
-/obj/structure/closet/toolcloset/Initialize()
-	. = ..()
-	if(prob(40))
-		new /obj/item/clothing/suit/storage/hazardvest(src)
-	if(prob(70))
-		new /obj/item/flashlight(src)
-	if(prob(70))
-		new /obj/item/screwdriver(src)
-	if(prob(70))
-		new /obj/item/wrench(src)
-	if(prob(70))
-		new /obj/item/weldingtool(src)
-	if(prob(70))
-		new /obj/item/crowbar(src)
-	if(prob(70))
-		new /obj/item/wirecutters(src)
-	if(prob(70))
-		new /obj/item/t_scanner(src)
-	if(prob(20))
-		new /obj/item/storage/belt/utility(src)
-	if(prob(30))
-		new /obj/item/stack/cable_coil/random(src)
-	if(prob(30))
-		new /obj/item/stack/cable_coil/random(src)
-	if(prob(30))
-		new /obj/item/stack/cable_coil/random(src)
-	if(prob(20))
-		new /obj/item/multitool(src)
-	if(prob(5))
-		new /obj/item/clothing/gloves/insulated(src)
-	if(prob(40))
-		new /obj/item/clothing/head/hardhat(src)
+/obj/structure/closet/firecloset/chief/WillContain()
+	return list(
+		new /datum/atom_creator/simple(/obj/item/clothing/suit/storage/hazardvest, 40),
```

Comment:
What do you mean? The atom_creator?

Reply:
This might clear things up:
```
/proc/create_objects_in_loc(var/atom/loc, var/atom_paths)
	if(!istype(loc))
		CRASH("Inappropriate loction given.")

	if(istype(atom_paths, /datum/atom_creator))
		var/datum/atom_creator/atom_creator = atom_paths
		atom_creator.create(loc)
	else if(islist(atom_paths))
		for(var/atom_path in atom_paths)
			for(var/i = 1 to max(1, atom_paths[atom_path]))
				create_objects_in_loc(loc, atom_path)
	else if(ispath(atom_paths))
		new atom_paths(loc)
	else
		CRASH("Unhandled input: [log_info_line(atom_paths)]")
```